lib/foreman/engine.rb

```diff
@@ -369,28 +363,28 @@ def spawn_processes
     end
   end
 
-  def read_self_pipe
```

Comment:
I'm not actually sure what the purpose of this was. Presumably we're not expecting (printable) output on @selfpipe, because name_for(@readers.invert[reader]) would have failed anyway.